### ISSUE TITLE
[DOCS] Misleading docs : reactive-forms#revert-cancel-changes

### ISSUE THREAD

**Issue by seedy** [angular/angular#21968]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Feature request
[x] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

Code is not doing what docs say : 
https://angular.io/guide/reactive-forms#revert-cancel-changes

Clicking the revert button triggers a call to the component's revert method. Users now can save or revert changes.

Actually (from w3c docs : https://www.w3schools.com/tags/att_button_type.asp) clicking a button with type="reset" "resets the form-data to its initial values".

Therefore the revert() function is useless and leads to a bug which took me 1h to resolve because I am performing my own handling of form in revert method.

## Expected behavior

Do not tell people to use a button with type reset when you want to perform your own behaviour on click. Because it is very likely to get overriden by default html5 behaviours.

## Minimal reproduction of the problem with instructions
Try demo : https://nvvryrkmnnob.angular.stackblitz.io/
<button type="reset" (click)="revert()"
            [disabled]="heroForm.pristine" class="btn btn-danger">Revert</button>

## What is the motivation / use case for changing the behavior?

Erroneous guide.

Browser:
- [x] Chrome (desktop) version 63.0.3239.132
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX

**Comment by bzmillerboy** [comment]
I agree this seems incorrect, however in my case, using `type="reset"` helped prevent an unwanted update to the form data when the user clicked my "Cancel" button that called ```  goBack() {
    this.location.back();
  }``` .

Therefore, how might I better revert changes to the form and take the user back to the previous screen when they cancel? I've created a "Cancel Button" component to perform the `goBack` method above.

My `BackButtonComponent` doesn't know about `this.ngOnChanges();`.

**Comment by bzmillerboy** [comment]
How might I reset the form if the "Reset" button is it's own component that simply calls ```  goBack() {
    this.location.back();
  }``` ?

In my case, adding `type=reset` helped prevent the form from sending an update to the DB when the user click "Reset" button.

**Comment by kapunahelewong** [comment]
@kara would you have any advice on the @bzmillerboy added? I'm not sure (but interested!).

**Comment by seedy** [comment]
@bzmillerboy In a case where you need the basic behaviour from a reset button, just stick to type="reset"
My goal here is to make sure that the consequences of setting type="reset" are understood by beginners.

Here's an example of what I'm doing with my reset button : 
(html)
```
<button md-raised-button color="warn" (click)="reverted()">
```

(ts)
```
@Output() revert = new EventEmitter<void>();

public reverted(): void {
        this.revert.emit();
    }
```

Parent component is taking care of his form when he catches the event. 
I am working with edit forms, therefore when I reset, I actually reset to the data's state before form was dirtied.

**Comment by jenniferfell** [comment]
@kapunahelewong : It looks like this was resolved by #22094, yes? 
https://angular.io/guide/reactive-forms#revert-cancel-changes
https://angular.io/guide/reactive-forms#buttons

If there's more work to do, let's connect with @brandonroberts . Thanks!

**Comment by kapunahelewong** [comment]
Yep, @jenniferfell this was resolved in that PR. Closing.
